parmoo/surrogates/polynomial.py

```diff
@@ -0,0 +1,322 @@
+
+""" Implementations of the SurrogateFunction class.
+
+This module contains implementations of the SurrogateFunction ABC, which rely
+on local linear models.
+
+The classes include:
+ * ``Linear`` -- fits a local linear surrogate
+
+"""
+
+import numpy as np
+from scipy.stats import tstd
+from parmoo.structs import SurrogateFunction
+from parmoo.util import xerror
+
+
+class Linear(SurrogateFunction):
+    """ A local linear surrogate model.
+
+    This class implements a local (TR constrained) linear surrogate,
+    using the SurrogateFunction ABC.
+
+    """
+
+    # Slots for the UniformRandom class
+    __slots__ = ['m', 'n', 'lb', 'ub', 'x_vals', 'f_vals', 'eps',
+                 'n_loc', 'loc_inds', 'tr_center', 'rad', 'weights',
+                 'prev_centers']
+
+    def __init__(self, m, lb, ub, hyperparams):
+        """ Constructor for the Linear class.
+
+        Args:
+            m (int): The number of objectives to fit.
+
+            lb (numpy.ndarray): A 1d array of lower bounds for the design
+                region. The number of design variables is inferred from the
+                dimension of lb.
+
+            ub (numpy.ndarray): A 1d array of upper bounds for the design
+                region. The dimension must match ub.
+
+            hyperparams (dict): A dictionary of hyperparameters for the
+                RBF models, including:
+                 * des_tols (numpy.ndarray, optional): A 1d array whose length
+                   matches lb and ub. Each entry is a number (greater than 0)
+                   specifying the design space tolerance for that variable.
+                   By default, des_tols = [1.0e-8, ..., 1.0e-8].
```

Comment:
Confirm whether this default is still precise

Reply:
Same as above